Library/Homebrew/test/formulary_spec.rb

```diff
@@ -0,0 +1,215 @@
+require "formula"
+require "formula_installer"
+require "utils/bottles"
+
+describe Formulary do
+  let(:name) { "testball_bottle" }
+  let(:path) { CoreTap.new.formula_dir/"#{name}.rb" }
+  let(:content) {
+    <<-EOS.undent
+      class #{described_class.class_s(name)} < Formula
+        url "file://#{TEST_FIXTURE_DIR}/tarballs/testball-0.1.tbz"
+        sha256 TESTBALL_SHA256
+
+        bottle do
+          cellar :any_skip_relocation
+          root_url "file://#{bottle_dir}"
+          sha256 "9abc8ce779067e26556002c4ca6b9427b9874d25f0cafa7028e05b5c5c410cb4" => :#{Utils::Bottles.tag}
+        end
+
+        def install
+          prefix.install "bin"
+          prefix.install "libexec"
+        end
+      end
+    EOS
+  }
+  let(:bottle_dir) { Pathname.new("#{TEST_FIXTURE_DIR}/bottles") }
+  let(:bottle) { bottle_dir/"testball_bottle-0.1.#{Utils::Bottles.tag}.bottle.tar.gz" }
+
+  specify "::class_s" do
+    expect(described_class.class_s("shell.fm")).to eq("ShellFm")
+    expect(described_class.class_s("foo++")).to eq("Fooxx")
+    expect(described_class.class_s("s-lang")).to eq("SLang")
+    expect(described_class.class_s("pkg-config")).to eq("PkgConfig")
+    expect(described_class.class_s("foo_bar")).to eq("FooBar")
+    expect(described_class.class_s("openssl@1.1")).to eq("OpensslAT11")
+  end
+
+  describe "::factory" do
+    before(:each) do
+      path.write content
+    end
+
+    it "returns a Formula" do
+      expect(described_class.factory(name)).to be_kind_of(Formula)
+    end
+
+    it "returns a Formula when given a fully qualified name" do
+      expect(described_class.factory("homebrew/core/#{name}")).to be_kind_of(Formula)
+    end
+
+    it "raises an error if the Formula cannot be found" do
+      expect {
+        described_class.factory("not_existed_formula")
+      }.to raise_error(FormulaUnavailableError)
+    end
+
+    context "if the Formula has the wrong class" do
+      let(:name) { "giraffe" }
```

Comment:
At first I thought you must be right, but since the class is evaluated before with the first name, we need to use a separate name for this specific test.